geniesp/bpc_redcap_export_mapping.py

```diff
@@ -579,9 +597,74 @@ def check_oncotree_codes(
             f"There are invalid values in ONCOTREE_CODE column in the clinical df: {invalid_codes}")
 
 
+def replace_cpt_seq_date(
+    input_data : pd.DataFrame, 
+    replacement_data : pd.DataFrame,
+    cpt_seq_date_replacement_type : str
+    ) -> pd.DataFrame:
+    """ Replaces CPT_SEQ_DATE in input BPC file with specified replacement data's 
+        CPT_SEQ_DATE because the values in CPT_SEQ_DATE are incorrect.
+        
+        Current available cpt_seq_date_replacement_type values are ["main_genie", "derived_variable"]
+            - main_genie - uses the main genie consortium release's sample file and 
+            - derived_variable - uses the stats team's provided derived variable file
+                after BPC processing is complete and tables are provided
+
+    Args:
+        input_data (pd.DataFrame): input data with CPT_SEQ_DATE values to be replaced
+        replacement_data (pd.DataFrame): the derived variable data with CPT_SEQ_DATE values to 
+            use as replacement
+        cpt_seq_date_replacement_type (str): the type of replacement used, (e.g: main genie, derived variable)
+    
+    Raises:
+        ValueError: thrown when cpt_seq_date_replacement_type is not one of the valid
+            allowed values
+    
+    Returns:
+        pd.DataFrame: input data with replaced CPT_SEQ_DATE values
+    """
+    # Remove CPT_SEQ_DATE because the values are incorrect
+    del input_data["CPT_SEQ_DATE"]
+    logging.info(f"Replacing CPT_SEQ_DATE values with {cpt_seq_date_replacement_type} values ...")
+    if cpt_seq_date_replacement_type == "main_genie":
+        
+        # main genie clinical is just sample data
+        replacement_data = replacement_data[["SAMPLE_ID", "SEQ_YEAR"]]
+        replacement_data.rename(columns={"SEQ_YEAR": "CPT_SEQ_DATE"}, inplace=True)
+        merge_cols = ["SAMPLE_ID"]
+        
+    elif cpt_seq_date_replacement_type == "derived_variable":
+        # there should only be a unique seq_date per patient id and sample id
+        replacement_data = replacement_data[
+            ["cpt_genie_sample_id", "record_id", "cpt_seq_date"]
```

Comment:
Nit. As the investigation attached below, I believe that the `cpt_genie_sample_id` should be good enough to be the unique identifiers used for merging. 

<img width="1074" alt="Screenshot 2025-04-09 at 4 28 01 PM" src="https://github.com/user-attachments/assets/6e9439d8-44e5-4aa6-93dc-dce2ca49e3e5" />

Reply:
I think that makes sense since `cpt_seq_date` is cancer panel test specific variable thus `sample` specific variable so there should only be a unique `cpt_seq_date` per `cpt_genie_sample_id`